### PR TITLE
Add parser interface with base implementation and single column singl…

### PR DESCRIPTION
**Pull request by MortalFlesh**
First of all - sorry for more than 500 changes, but this is the smallest set of changes on adding a parser interaface.

More parsers will come later to cover all parsing cases in `QueryParameterParser`.

### CODE REVIEW

**Comment by ghost** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR is way too big, consider splitting it up to smaller ones. Reviewers will be thankful 🙏.</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/Service/Parser/SingleColumnSingleValueParser.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace Lmc\ApiFilter\Service\Parser;
+
+use Lmc\ApiFilter\Constant\Filter;
+
+class SingleColumnSingleValueParser extends AbstractParser
+{
+    public function supports(string $rawColumn, $rawValue): bool
+    {
+        return is_string($rawColumn) && !$this->isTuple($rawColumn) && !is_array($rawValue) && !$this->isTuple($rawValue);
```
Comment (legendik) [inline src/Service/Parser/SingleColumnSingleValueParser.php]: `is_string(rawColumn)` evaluates always to string. I think you can remove it.

File: tests/Filters/FiltersTest.php [hunk 2]
```diff
@@ -107,4 +108,33 @@ public function shouldCountFilters(): void
...
+        $filter1 = new FilterWithOperator('col1', new Value('val'), '=', 'eq');
+        $filter2 = new FilterWithOperator('col2', new Value('val'), '=', 'eq');
+        $filter3 = new FilterWithOperator('col3', new Value('val'), '=', 'eq');
+
+        return [
+            // filters, filter, expected
+            'empty filters' => [[], $filter1, false],
+            'has filter exactly one' => [[$filter1], $filter1, true],
+            'has filter from more' => [[$filter1, $filter2], $filter2, true],
+            'has not filter exactly one' => [[$filter1], $filter2, false],
```
Comment (legendik) [inline tests/Filters/FiltersTest.php]: `has not filter from one`?

File: tests/Service/Parser/Fixtures/DummyFilter.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\ApiFilter\Service\Parser\Fixtures;
+
+use Lmc\ApiFilter\Entity\Value;
+use Lmc\ApiFilter\Filter\FilterInterface;
+
+class DummyFilter implements FilterInterface
```
Comment (legendik) [inline tests/Service/Parser/Fixtures/DummyFilter.php]: It is not Dummy at all. It is using for tests. How about `TestingFilter`, `RawFilter`, `SimpleFilter` or somethink else?

**Review by legendik — CHANGES_REQUESTED** [review]

**Comment by legendik** [comment]
Good work BTW 🎉

File: src/Service/Parser/SingleColumnSingleValueParser.php (hunk 1 above)
Comment (MortalFlesh) [reply src/Service/Parser/SingleColumnSingleValueParser.php]: good point, it was there, because from the beginning it was `mixed`

**Review by legendik — APPROVED** [review]

**Pull request merged**